src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -45,6 +45,7 @@ public function __construct(array $functions)
         $this->binaryOperators = [
             'or' => ['precedence' => 10, 'associativity' => self::OPERATOR_LEFT],
             '||' => ['precedence' => 10, 'associativity' => self::OPERATOR_LEFT],
+            'xor' => ['precedence' => 11, 'associativity' => self::OPERATOR_LEFT],
```

Comment:
should it really have higher precedence than `||`? 

in php the expression `1 xor 0 || 1` will evaluate to `false` because `xor` has lower precedence than `||`: http://php.net/manual/fa/language.operators.precedence.php

I think with your precedence it will evaluate to `true` when used in an expression?